test/autoinst_profile/partition_section_test.rb

```diff
@@ -0,0 +1,297 @@
+      expect(section_for("sdb1").size).to eq Y2Storage::DiskSize.GiB(780).to_i.to_s
+    end
+
+    context "if the partition contains a filesystem" do
+      it "initializes #filesystem with the corresponding symbol" do
+        expect(section_for("dasdb1").filesystem).to eq :swap
+        expect(section_for("dasdb2").filesystem).to eq :btrfs
+        expect(section_for("dasdb3").filesystem).to eq :xfs
+      end
+
+      it "initializes #label to the filesystem label" do
+        expect(section_for("dasdb2").label).to eq "suse_root"
+      end
+
+      it "initializes #label to nil if the filesystem has no label" do
+        expect(section_for("dasdb3").label).to be_nil
+      end
+
+      context "if the filesystem contains mounting information" do
+        it "initializes #mount and #mountby" do
+          section = section_for("sdc3")
+          expect(section.mount).to eq "/"
+          expect(section.mountby).to eq :uuid
+        end
+      end
+
+      context "if the filesystem is not configured to be mounted" do
+        it "initializes #mount and #mountby to nil" do
+          section = section_for("dasdb2")
+          expect(section.mount).to be_nil
+          expect(section.mountby).to be_nil
+        end
+      end
+
+      it "initializes #format to true for most partition ids" do
+        expect(section_for("sdg1").format).to eq true
+        expect(section_for("sdg4").format).to eq true
+      end
+
+      # Weird logic inherited from the old code
+      it "initializes #format to false for PReP and DOS16 partitions" do
+        expect(section_for("sdg2").format).to eq false
+        expect(section_for("sdg3").format).to eq false
+      end
+    end
+
+    context "if the partition contains no filesystem" do
+      before { allow_any_instance_of(Y2Storage::Partition).to receive(:filesystem).and_return nil }
+
+      it "initializes #filesystem, #label, #mount and #mountby to nil" do
+        expect(section_for("sdc3").filesystem).to be_nil
+        expect(section_for("sdc3").label).to be_nil
+        expect(section_for("sdc3").mount).to be_nil
+        expect(section_for("sdc3").mountby).to be_nil
+      end
+
+      it "initializes #format to false despite the partition id" do
+        expect(section_for("sdg1").format).to eq false
+        expect(section_for("sdg2").format).to eq false
+        expect(section_for("sdg3").format).to eq false
+        expect(section_for("sdg4").format).to eq false
+      end
+    end
+
+    context "if the partition is encrypted" do
+      it "initializes #crypt_key to a generic string" do
+        expect(section_for("sdf7").crypt_key).to eq "ENTER KEY HERE"
+      end
+
+      it "initializes #loop_fs and #crypt_key to true" do
+        section = section_for("sdf7")
+        expect(section.crypt_fs).to eq true
+        expect(section.loop_fs).to eq true
+      end
+    end
+
+    context "if the partition is not encrypted" do
+      it "initializes #crypt_key to a generic string" do
```

Comment:
Just curiosity: why not to `nil` or an empty string? I guess it's something from the past...